### PR TITLE
Add jaxws support

### PR DESCRIPTION
**Pull request by mattdrees**
This PR adds jax-ws support.

This was originally  intended to be used by CCP, and hopefully some day that will happen, but currently there are other issues that are preventing red egg from being used by CCP.

### CODE REVIEW

**Comment by mattdrees** [comment]
@hlbraddock can you review this for me? I want to use some of the changes here in the next PR (the one you are interested in.)

**Comment by hlbraddock** [comment]
Does red-egg still provide no cdi app support?

**Comment by mattdrees** [comment]
Yes, it supports apps w/out cdi

**Comment by mattdrees** [comment]
perhaps not for @Action though

File: src/main/java/org/cru/redegg/recording/impl/DefaultWebErrorRecorder.java [hunk 1]
```diff
@@ -137,7 +142,9 @@ public WebErrorRecorder recordRequestPostParameters(Multimap<String, String> pos
     @Override
     public WebErrorRecorder recordEntityRepresentation(String entityRepresentation) {
         checkState(!completed);
-        webContext.setEntityRepresentation(checkNotNull(entityRepresentation));
+        checkNotNull(entityRepresentation);
+        String sanitized = entitySanitizer.sanitizeEntity(entityRepresentation);
+        webContext.setEntityRepresentation(sanitized);
```
Comment (hlbraddock) [inline src/main/java/org/cru/redegg/recording/impl/DefaultWebErrorRecorder.java]: Regarding `DefaultWebErrorRecorder`, line 161 in this PR, I realize there is no change to this code in this PR, but I'm confused that you should be doing `checkState(!completed)` in a method where you are setting the completed state. Can you clarify for me?

**Comment by hlbraddock** [comment]
You might include documentation on github repository front page for how to provide custom entity sanitizer for SOAP apps.

**Comment by hlbraddock** [comment]
Have you written any tests ensuring the entity has been sanitized?

**Comment by hlbraddock** [comment]
I have no other comments to add.

File: src/main/java/org/cru/redegg/recording/impl/DefaultWebErrorRecorder.java (hunk 1 above)
Comment (mattdrees) [inline src/main/java/org/cru/redegg/recording/impl/DefaultWebErrorRecorder.java]: It means you can't call the method twice.

Comment (mattdrees) [inline src/main/java/org/cru/redegg/recording/impl/DefaultWebErrorRecorder.java]: Or maybe I'm misunderstanding your question.

**Comment by mattdrees** [comment]
Looks like I forgot to add a test for entity sanitization. I will add one.

**Comment by hlbraddock** [comment]
OK. I've read your comments. :+1: 

Once you've added the entity sanitizer tests, this PR should be good to go.

**Comment by mattdrees** [comment]
Ok, I think I've addressed everything. I added a test, fixed a few related problems, and updated the docs. Let me know what you think.

**Comment by hlbraddock** [comment]
You might also want to release a new version as I believe this will break the build of ERT, due to https://github.com/CruGlobal/conf-registration-api/blob/master/src/test/java/org/cru/crs/tests/ResourceTestClassCollection.java

**Comment by mattdrees** [comment]
Good point.. I will do so.

However, I don't understand that implementation. It appears as if that `NullWebErrorRecorder` cannot be used. (It's private, and not instantiated in that file.) So, I think it could be removed.

**Comment by hlbraddock** [comment]
It's being made available to Archillian, I believe.

**Comment by hlbraddock** [comment]
Though it may not be able to access it, being private. You can talk to the author of that page for more info. I presume it was @ryancarlson

**Comment by mattdrees** [comment]
ok, version is bumped.

**Comment by mattdrees** [comment]
all good?

**Comment by hlbraddock** [comment]
All good! :+1:

**Comment by mattdrees** [comment]
doh, bumping the version broke a test. Will have to come up with a solution.

**Comment by mattdrees** [comment]
Ok, think this is solved. Look ok?

**Comment by hlbraddock** [comment]
Looks good.

**Comment by mattdrees** [comment]
thanks again!

**Comment by mattdrees** [comment]
I forgot a file in my last commit. :-(

I've fixed in master, in b00b94ce3efe718311d920e349f6a9b5042a3dfe.

**Pull request merged**